internal/controller/processor.go

```diff
@@ -270,6 +283,9 @@ func (processor *AppConfigurationProviderProcessor) shouldReconcile(
 
 func (processor *AppConfigurationProviderProcessor) Finish() (ctrl.Result, error) {
 	processor.ReconciliationState.Generation = processor.Provider.Generation
+	if processor.RefreshOptions.SecretSettingPopulated {
```

Comment:
Is there any sepcial reason of not putting this block into the line295?

Reply:
Not all reconcile triggered by refresh, when `shouldReconcile=true` such as delete configmap/secret, we need to maintain `existingSecretReferences` when `secret` section specified.